Source/CesiumRuntime/Private/ScreenCreditsWidget.cpp

```diff
@@ -0,0 +1,241 @@
+    FString url;
+    FString text;
+    const FSlateBrush* Brush = nullptr;
+    if (RunInfo.MetaData.Contains(TEXT("url"))) {
+      url = *RunInfo.MetaData[TEXT("url")];
+    }
+    if (RunInfo.MetaData.Contains(TEXT("text"))) {
+      text = *RunInfo.MetaData[TEXT("text")];
+    }
+    if (RunInfo.MetaData.Contains(TEXT("id"))) {
+      int32 id = FCString::Atoi(*RunInfo.MetaData[TEXT("id")]);
+      Brush = Decorator->FindImageBrush(id);
+    }
+    return SNew(SRichInlineImage, Brush, url, text, TextStyle, Decorator);
+  }
+
+private:
+  UCreditsDecorator* Decorator;
+};
+
+UCreditsDecorator::UCreditsDecorator(
+    const FObjectInitializer& ObjectInitializer)
+    : URichTextBlockDecorator(ObjectInitializer) {}
+
+TSharedPtr<ITextDecorator>
+UCreditsDecorator::CreateDecorator(URichTextBlock* InOwner) {
+  return MakeShareable(new FRichInlineImage(InOwner, this));
+}
+
+const FSlateBrush* UCreditsDecorator::FindImageBrush(int32 id) {
+  if (CreditsWidget->_creditImages.Num() > id) {
+    return CreditsWidget->_creditImages[id];
+  }
+  return nullptr;
+}
+
+UScreenCreditsWidget::UScreenCreditsWidget(
+    const FObjectInitializer& ObjectInitializer)
+    : UUserWidget(ObjectInitializer) {
+  static ConstructorHelpers::FObjectFinder<UFont> RobotoFontObj(
+      *UWidget::GetDefaultFontName());
+  _font = FSlateFontInfo(RobotoFontObj.Object, 8);
+}
+
+void UScreenCreditsWidget::OnPopupClicked() {
+  _showPopup = !_showPopup;
+  BackgroundBlur->SetVisibility(
+      _showPopup ? ESlateVisibility::Visible : ESlateVisibility::Hidden);
+}
+
+void UScreenCreditsWidget::NativeConstruct() {
+
+  Super::NativeConstruct();
+
+  if (RichTextOnScreen) {
+    RichTextOnScreen->SetDefaultFont(_font);
+    RichTextOnScreen->SetDefaultColorAndOpacity(
+        FSlateColor(FLinearColor(1.f, 1.f, 1.f, 1.f)));
+    _decoratorOnScreen =
+        static_cast<UCreditsDecorator*>(RichTextOnScreen->GetDecoratorByClass(
+            UCreditsDecorator::StaticClass()));
+
+    _decoratorOnScreen->PopupClicked.BindUObject(
+        this,
+        &UScreenCreditsWidget::OnPopupClicked);
+    _decoratorOnScreen->CreditsWidget = this;
+  }
+  if (RichTextPopup) {
+    RichTextPopup->SetDefaultFont(_font);
+    RichTextPopup->SetDefaultColorAndOpacity(
+        FSlateColor(FLinearColor(1.f, 1.f, 1.f, 1.f)));
+    _decoratorPopup = static_cast<UCreditsDecorator*>(
+        RichTextPopup->GetDecoratorByClass(UCreditsDecorator::StaticClass()));
+
+    _decoratorPopup->CreditsWidget = this;
+  }
+}
+
+void UScreenCreditsWidget::HandleImageRequest(
+    FHttpRequestPtr HttpRequest,
+    FHttpResponsePtr HttpResponse,
+    bool bSucceeded,
+    int32 id) {
+  if (bSucceeded && HttpResponse.IsValid() &&
+      HttpResponse->GetContentLength() > 0) {
+    UTexture2D* texture =
+        FImageUtils::ImportBufferAsTexture2D(HttpResponse->GetContent());
+    texture->SRGB = true;
+    texture->UpdateResource();
+    texture->AddToRoot();
```

Comment:
I don't think this AddToRoot (or the other one below) is needed as long as you make `_creditImages` a `UPROPERTY`. And that'll avoid a memory leak, too, because there's no RemoveFromRoot anywhere.